api/geth_backend.go

```diff
@@ -2584,6 +2587,14 @@ func (b *GethStatusBackend) Logout() error {
 	return nil
 }
 
+func (b *GethStatusBackend) switchToPreLoginLog() error {
```

Comment:
I think it's beneficial to add a comment somewhere (could be here as a function documentation) explaining why this pre-login approach exists, otherwise the rationale will be a scattered and difficult to find in GH discussions.